code/5448-slobodanka-lozanovska/script.ts

```diff
@@ -0,0 +1,273 @@
+      return first.series - second.series;
+    });
+    displayEpisodes(data);
+  }
+  
+  const sortByEra = () => {
+    const eraSort = document.getElementById("sort-era")!;
+  
+    currentSort = "era";
+    const allSorters = document.getElementsByClassName("sorter");
+    for (let index = 0; index < allSorters.length; index++) {
+      const sorter = allSorters[index];
+      sorter.classList.remove("sorted");
+      sorter.classList.add("unsorted");
+    }
+    eraSort.classList.toggle("sorted");
+    eraSort.classList.toggle("unsorted");
+  
+    data.sort((first: Episode, second: Episode) => {
+      return first.era.localeCompare(second.era);
+    });
+    displayEpisodes(data);
+  }
+  
+  const sortByBroadcast = () => {
+    const broadcastSort = document.getElementById("sort-broadcast")!;
+  
+    currentSort = "broadcast_date";
+    const allSorters = document.getElementsByClassName("sorter");
+    for (let index = 0; index < allSorters.length; index++) {
+      const sorter = allSorters[index];
+      sorter.classList.remove("sorted");
+      sorter.classList.add("unsorted");
+    }
+    broadcastSort.classList.toggle("sorted");
+    broadcastSort.classList.toggle("unsorted");
+  
+    data.sort((first: Episode, second: Episode) => {
+      return first.broadcast_date.localeCompare(second.broadcast_date);
+    });
+    displayEpisodes(data);
+  }
+  
+  const sortByDirector = () => {
+    const directorSort = document.getElementById("sort-director")!;
+  
+    currentSort = "director";
+    const allSorters = document.getElementsByClassName("sorter");
+    for (let index = 0; index < allSorters.length; index++) {
+      const sorter = allSorters[index];
+      sorter.classList.remove("sorted");
+      sorter.classList.add("unsorted");
+    }
+    directorSort.classList.toggle("sorted");
+    directorSort.classList.toggle("unsorted");
+  
+    data.sort((first: Episode, second: Episode) => {
+      return first.director.localeCompare(second.director);
+    });
+    displayEpisodes(data);
+  }
+  
+  const sortByWriter = () => {
+    const writerSort = document.getElementById("sort-writer")!;
+  
+    currentSort = "writer";
+    const allSorters = document.getElementsByClassName("sorter");
+    for (let index = 0; index < allSorters.length; index++) {
+      const sorter = allSorters[index];
+      sorter.classList.remove("sorted");
+      sorter.classList.add("unsorted");
+    }
+    writerSort.classList.toggle("sorted");
+    writerSort.classList.toggle("unsorted");
+  
+    data.sort((first: Episode, second: Episode) => {
+      return first.writer.localeCompare(second.writer);
+    });
+    displayEpisodes(data);
+  }
+  
+  const searchEpisodes = () => {
+    const episodeSearch = document.getElementById("name-filter")! as HTMLInputElement;
+    const episodeValue = episodeSearch.value.toLowerCase();
+
+    const eraSearch = document.getElementById("era-filter")! as HTMLSelectElement;
```

Comment:
The data in the filter is never properly initialized, so only search by name is operational